src/core/server/opensearch_data/opensearch_data_service.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+import { Observable, Subject } from 'rxjs';
+import { first, map, shareReplay, takeUntil } from 'rxjs/operators';
+import { CoreService } from 'src/core/types';
+import { AuditorFactory, AuditTrailStart } from '../audit_trail';
+import { CoreContext } from '../core_context';
+import { Logger } from '../logging';
+import { OpenSearchClientConfig } from '../opensearch/client';
+import { OpenSearchConfig, OpenSearchConfigType } from '../opensearch/opensearch_config';
+import { pollOpenSearchNodesVersion } from '../opensearch/version_check/ensure_opensearch_version';
+import { InternalSavedObjectsServiceStart, SavedObjectsClient } from '../saved_objects';
+import { SavedObjectsClientContract } from '../types';
+import { DataSourceClient } from './client/data_source_client';
+import { InternalOpenSearchDataServiceSetup, InternalOpenSearchDataServiceStart } from './types';
+
+interface StartDeps {
+  savedObjects: InternalSavedObjectsServiceStart;
+  auditTrail: AuditTrailStart;
+}
+
+export class OpenSearchDataService
+  implements CoreService<InternalOpenSearchDataServiceSetup, InternalOpenSearchDataServiceStart> {
+  private readonly log: Logger;
+  private readonly config$: Observable<OpenSearchConfig>;
+  private auditorFactory?: AuditorFactory;
+  private stop$ = new Subject();
```

Comment:
how is this field being used?

Reply:
it's used in the Compatibility check(that I commented out for now). Compatibility checks happen constantly until "this.stop" is called (which it will be called in `stop()` stage in our case)